pkg/trickster/prometheus/client.go

```diff
@@ -0,0 +1,45 @@
+package prometheus
+
+import (
+	"net/http"
+
+	"github.com/Comcast/trickster/internal/cache/registration"
+	"github.com/Comcast/trickster/internal/config"
+	"github.com/Comcast/trickster/internal/proxy/origins/prometheus"
+	"github.com/go-kit/kit/log"
+)
+
+type PrometheusClient struct {
+	client *prometheus.Client
+}
+
+func NewPrometheusClient(originConfig *config.OriginConfig, logger log.Logger) (*PrometheusClient, error) {
+	c, err := registration.GetCache(originConfig.CacheName)
+	if err != nil {
+		return nil, err
+	}
+	t := &PrometheusClient{
+		client: prometheus.NewClient(originConfig.Type, originConfig, c, logger),
+	}
+	return t, nil
+}
+
+func (t *PrometheusClient) Query(w http.ResponseWriter, r *http.Request) {
```

Comment:
I was referring to a more parsed-out specific handler type e.g. https://godoc.org/github.com/prometheus/client_golang/api/prometheus/v1#API


This is an improvement, but this would mean that all library users would have to convert their parsed out types into http request/response objects just to use this library.